daemon/images/image.go

```diff
@@ -51,13 +51,13 @@ func (i *ImageService) PrepareSnapshot(ctx context.Context, id string, image str
 	panic("not implemented")
 }
 
-func (i *ImageService) manifestMatchesPlatform(ctx context.Context, img *image.Image, platform ocispec.Platform) (bool, error) {
+func (i *ImageService) manifestMatchesPlatform(ctx context.Context, img *image.Image, platform ocispec.Platform) bool {
 	logger := log.G(ctx).WithField("image", img.ID).WithField("desiredPlatform", platforms.Format(platform))
 
 	ls, leaseErr := i.leases.ListResources(ctx, leases.Lease{ID: imageKey(img.ID().String())})
 	if leaseErr != nil {
 		logger.WithError(leaseErr).Error("Error looking up image leases")
-		return false, leaseErr
```

Comment:
Context cancellation will also result in the platform mismatch error.
How about leaving the function signature as is, and just change this line to return `nil` error?

Reply:
Also, we should handle context cancellation in the loop below (in a follow-up, as it's not strictly related to the subject of this PR).

Reply:
Okay changed it to return `false, nil` if the lease is not found and to return the error otherwise